src/actions/action.js

```diff
@@ -11,17 +11,30 @@ export default function action(type, host, path, params = {}, transform) {
       type: `OK/${type}`,
       payload,
     });
+    const getError = error => ({
+      type: `ERROR/${type}`,
+      error,
+    });
     const fetchDataWithRetry = delay => fetch(url, path === 'api/metadata' ? { credentials: 'include' } : {})
       .then((response) => {
         if (!response.ok) {
-          throw new Error('fetch failed');
+          dispatch(getError(response.status));
+          if (response.status === 404) {
+            const err = new Error('api returned 404');
+            err.code = 404;
```

Comment:
probably err.code = response.status.  Technically all 4xx codes indicate user error and should not be retried.